crates/astria-account-monitor/src/account_monitor.rs

```diff
@@ -82,40 +98,70 @@ impl AccountMonitor {
     ///
     /// # Errors
     /// An error is returned if bridge last transaction height is not found.
-    pub async fn run(&self) -> eyre::Result<()> {
-        let shutdown_token = ShutdownHandle::new();
-        let Some(()) = shutdown_token
-            .token
-            .run_until_cancelled(run_loop(
+    async fn run(self) -> eyre::Result<()> {
+        let mut poll_timer = interval(self.interval);
+        poll_timer.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
+
+        loop {
+            if self.shutdown_token.is_cancelled() {
+                info!("received shutdown signal");
+                break;
+            }
+
+            poll_timer.tick().await;
+
+            fetch_all_info(
                 self.metrics,
                 &self.sequencer_abci_client,
                 &self.sequencer_accounts,
                 &self.sequencer_asset,
-                self.interval,
-            ))
-            .await
-        else {
-            return Ok(());
-        };
+            );
+        }
 
         Ok(())
     }
 }
 
-async fn run_loop(
-    metrics: &'static Metrics,
-    client: &sequencer_client::HttpClient,
-    accounts: &SequencerAccountsToMonitor,
-    asset: &Asset,
-    pull_interval: Duration,
-) {
-    let mut poll_timer = interval(pull_interval);
-    poll_timer.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
+impl AccountMonitor {
```

Comment:
Group the impl blocks appropriately: `impl AccountMonitor` should be placed after its definition; all of its trait impls after the inherent impls.

Same for `Inner`